util/ranger/ranger.go

```diff
@@ -91,43 +91,54 @@ func points2Ranges(sc *stmtctx.StatementContext, rangePoints []point, tp *types.
 	return ranges, nil
 }
 
+// convertPoint converts this point according to the specified type.
+// isLeft is used to specify if this point is the left point in a range.
+// for any range (leftPoint, rightPoint),
+//		if the leftPoint is larger than the upperBound of this type, or
+//		the rightPoint is less than the lowerBound of this type,
+//		we think it's an invalid range and return with invalid=true.
+//	Here are some examples with the bound range of this type is [10, 20]:
```

Comment:
If we use type `tinyint` here to make this example, it'll be easier to understand.

Reply:
Updated, PTAL.